### PR TITLE
Adds required field validation for connection creation form

### PR DESCRIPTION
**Pull request by antony-britto**
# Adds required field validation for connection creation form

## Description
Adds required field validation for connection creation form

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [19004](https://cdap.atlassian.net/browse/CDAP-19004)

## Screenshots
image

### CODE REVIEW

File: app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx [hunk 1]
```diff
@@ -58,6 +60,33 @@ export function ConnectionConfigForm({
...
+  const [configurationErrors, setConfigurationErrors] = React.useState(
+    testResults.configurationErrors
+  );
+
+  const onCreate = () => {
+    const fieldErrors = {};
+    for (const [fieldName, value] of Object.entries(connectorProperties)) {
+      if (
+        !!value.required &&
+        (!values.hasOwnProperty(fieldName) || values[fieldName] === '' || values[fieldName] == null)
```
Comment (de-lan) [inline app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx:72]: should be `values[fieldName] === null` ?

Comment (antony-britto) [reply app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx:72]: intentionally covering both undefined and null

File: app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx [hunk 2]
```diff
@@ -58,6 +60,33 @@ export function ConnectionConfigForm({
   const [name, setName] = React.useState(initName);
   const [description, setDescription] = React.useState(initDescription);
   const classes = useStyle();
+  const [configurationErrors, setConfigurationErrors] = React.useState(
```
Comment (seanfeld) [inline app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx:63]: useState

**Review by seanfeld — APPROVED** [review]
just one nit

**Pull request merged**